client/blocks/edit-gravatar/index.jsx

```diff
@@ -254,13 +252,27 @@ export class EditGravatar extends Component {
 								/>
 							) }
 							<Gravatar imgSize={ GRAVATAR_IMG_SIZE } size={ 150 } user={ user } />
-							{ ! isUploading && (
-								<div className="edit-gravatar__label-container">
-									<Gridicon icon={ icon } size={ 36 } />
-									<span className="edit-gravatar__label">{ buttonText }</span>
+							<div className="edit-gravatar__label-container">
+								<div className="edit-gravatar__label-container-icon">
+									{ ! user.email_verified && (
+										<Icon className="gridicon" icon={ caution } fill="#fff" size={ 24 } />
+									) }
+
+									{ user.email_verified && ! isUploading && (
+										<Icon className="gridicon" icon={ upload } fill="#fff" size={ 24 } />
+									) }
+
+									{ user.email_verified && isUploading && (
+										<Spinner
+											style={ {
+												width: 24,
+												height: 24,
+											} }
+											className="edit-gravatar__label-container-icon-spinner"
+										/>
+									) }
```

Comment:
We should add some text to ensure there is some non-visual/accessible context.

It also looks like the `FilePicker` component (and the way it is used in this context) is not very accessible -- it's only `div` and `span` elements before the user clicks on the wrapper div, and we then show an unlabelled `input` field. But that may be something we can tackle in a follow-up.

Reply:
I am happy to iterate over the implementation in futher PRs to make it more accessible.

Adopting correct element roles and adding the proper aria-attributes can make way better.